kickstarter/android-oss
# NT-1442: Android Display an error state if add-ons fail to load

# 📲 What

Show an error alert dialog when there is an error retrieving the add-ons. Also allow user to tap retry to resend the networks calls to get the add-ons. 

# 🤔 Why

To give user more context on why add-ons are not loading correctly. 

# 🛠 How

Created a new stream that triggers the dialog when the network response is an error. Retry button resends the project data and reason to make a network call.

# 👀 See

![Screenshot_1600286472](https://user-images.githubusercontent.com/19390326/93387559-6f27f200-f837-11ea-8a9e-01cbe89fe0a6.png)

# 📋 QA

- Search for the project "one inch straw"
- Select "back project"
- Turn off wifi or put phone in airplane mode
- Select a pledge with add-ons
- Wait to see the error dialog
- Turn on wifi or turn off airplane mode
- Press "retry"
- After network call finishes, should show add-ons

# Story 📖

[NT-1442](https://kickstarter.atlassian.net/secure/RapidBoard.jspa?rapidView=9&projectKey=NT&modal=detail&selectedIssue=NT-1442&assignee=5efe5a70e0043f0bacaf7706&assignee=5eda9c57222d4d0b8296577e)
